upf/upf_pfcp_api.c

```diff
@@ -857,11 +915,99 @@ handle_create_pdr (upf_session_t * sx, pfcp_create_pdr_t * create_pdr,
 	create->pdi.nwi_index = nwi - gtm->nwis;
       }
 
+
+    vec_foreach (res, gtm->upip_res)
+    {
+      if (res->nwi_index == create->pdi.nwi_index)
+        break;
+    }
+
```

Comment:
upip_res is indexed (mhash) be nwi name. So instead of the loop, you could simply lookup it up the the NWI name.

Note: we divert from the PFCP specification a bit by always requiring the `Network Instance` to be present in PDIs. R16 also clarified that `NI` is required when a TEID CH bit set.